src/targets/gpu/fuse_ops.cpp

```diff
@@ -332,6 +337,48 @@ void move_standard_front(std::vector<instruction_ref>& args)
         std::swap(*it, args.front());
 }
 
+struct find_layernorm
+{
+    template <class... Ts>
+    static auto multibroadcast_op(Ts... xs)
+    {
+        return match::name("multibroadcast")(match::arg(0)(xs...));
+    }
+
+    static auto layernorm_onnx()
+    {
+        return match::arg(0)(match::name("gpu::sub")(
+            match::arg(0)(match::any().bind("x")),
+            match::arg(1)(multibroadcast_op(match::name("gpu::reduce_mean")))));
+    }
+
+    static auto layernorm_tf()
+    {
+        return match::either_arg(0, 1)(
+            match::name("gpu::mul")(match::arg(0)(match::name("gpu::add").bind("x"))),
+            match::name("gpu::sub")(match::either_arg(0, 1)(
+                multibroadcast_op(),
+                match::name("gpu::mul")(
+                    match::arg(1)(match::name("gpu::mul")(match::arg(1)(multibroadcast_op()))
+
+                                      )))));
```

Comment:
This never matches `reduce_mean`.  Does it need to be `multibroadcast_op(match::name("gpu::reduce_mean"))`?

Reply:
sorry I still have to add some more specific cases to the matcher that I missed, like you suggested